roles/mattermost/tasks/deploy_mattermost_operator.yml

```diff
@@ -0,0 +1,30 @@
+- name: Ensure Mattermost Operator Namespace Exists
+  community.kubernetes.k8s:
+    state: present
+    kubeconfig: "{{ kubeconfig }}"
+    api_version: v1
+    kind: Namespace
+    name: "{{ mattermost_namespace }}"
+
+- name: Subscribe to Mattermost Operator
+  community.kubernetes.k8s:
+    state: present
+    kubeconfig: "{{ kubeconfig }}"
+    definition: |-
+      apiVersion: operators.coreos.com/v1alpha1
+      kind: Subscription
+      metadata:
+        name: mattermost-operator
+        namespace: "{{ mattermost_namespace }}"
+      spec:
+        channel: "{{ mattermost_operator_channel }}"
+        installPlanApproval: Automatic
+        name: mattermost-operator
```

Comment:
Not a big fan of static specs. Please change name/source/sourceNameSpace into a variable. Also well as the meta data (seems the name are the same default values. 

Elsewise, you can also just default the variable in the task file itself if you don't want to extra into vars.

Reply:
name must/should be mattermost-operator for subscription, sourceNamespace should be openshift-marketplace, additonally  as requested I will change source name although no other operators do this (we should write a story on this stuff for the future, just as note)